x-pack/plugin/core/src/main/resources/ecs-template.yml

```
@@ -0,0 +1,222 @@
+    # As most fields are keyword, these will be automatically be matched correctly. Only ECS core fields were taken into account.
+    dynamic_templates:
+
+      ### Index mappings
+
+      # Message field is always index
+      # DISCUSS: Should only the top level message field always be indexed?
+      - match_message:
+          path_match: message
+          mapping:
+            type: match_only_text
+
+      ### Runtime mappings
+
+      # Set the ECS version
+      - match_ecs.version:
+          path_match: ecs.version
+          # DISCUSS: In ECS this is still a keyword field
+          type: version
+          # TODO: Can this somehow be set?
+          #value: ${xpack.stack.template.version}
+
+      # Matching all string fields by default to keyword
+      # As it is a runtime field, ignore_above is not needed
+      # TODO: Does first / last match win in the list?
+      - strings_as_keyword:
+          match_mapping_type: string
+          runtime:
+            type: keyword
+
+      # Matching all ECS core long fields
+      # This matches the following fields: bytes, packets, port, duration, severity, pid
+      # Path matching on these fields is needed as otherwise if the first value of the field would be for example
+      # "7" as a string in JSON, it would have the wrong default mapping. As it is a runtime field, the mapping could
+      # still be adjusted but not something we should have the user do by default.
+      # TODO: if path_match would be an array, this could be defined much nicer and more compact
+      - match_bytes:
+          path_match: bytes
+          runtime:
+            type: long
+      - match_packets:
+          path_match: packets
+          runtime:
+            type: long
+      - match_port:
+          path_match: port
+          runtime:
+            type: long
+      - match_duration:
+          path_match: duration
+          runtime:
+            type: long
+      - match_severity:
+          path_match: severity
+          runtime:
+            type: long
+      - match_pid:
+          path_match: pid
+          runtime:
+            type: long
+
+      # Matching all float and double fields
+      # This matches the following fields: event.risk_score
+      # Why path_match is used, see discussion for long. For runtime fields, it does not matter if it is a float or a double
+      - match_risk_score:
+          match_path: risk_score
+          runtime:
+            type: double
+
+      # Matching all boolean fields
+      # This matches the following fields: exists
+      # Why path_match is used, see discussion for long.
+      - match_exists:
+          match_path: exists
+          runtime:
+            type: boolean
+
+      # Match all ip fields to ip
+      # This matches the following fields: ip, network.forwarded_ip
+      - match_ip:
+          match: ip
+          runtime:
+            type: ip
+      - match_forwarded_ip:
+          match: forwarded_ip
```

If we anticipate more ip fields that end with `_ip`
```suggestion
          match: *_ip
```

---

I like the idea. Can't see a reason at the moment why something would be called _ip but not be an ip ...